src/tiled/templategroupdocument.cpp

```diff
@@ -46,10 +47,16 @@ TemplateGroupDocument::TemplateGroupDocument(TemplateGroup *templateGroup, const
     , mTemplateGroup(templateGroup)
 {
     mFileName = fileName;
+    // TODO: Tileset references are added with each template, if a tileset is not used it will be added again
+    // TilesetManager *tilesetManager = TilesetManager::instance();
+    // tilesetManager->addReferences(mTemplateGroup->tilesets());
 }
 
 TemplateGroupDocument::~TemplateGroupDocument()
 {
+    TilesetManager *tilesetManager = TilesetManager::instance();
+    tilesetManager->removeReferences(mTemplateGroup->tilesets());
```

Comment:
If you add references in the `TemplateGroup`, then please remove them there as well.